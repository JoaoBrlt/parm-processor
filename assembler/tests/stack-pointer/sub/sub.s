// Set the registers.
MOVS r0, #1
MOVS r1, #2
MOVS r2, #3
MOVS r3, #4
MOVS r4, #5
MOVS r5, #6
MOVS r6, #7
MOVS r7, #8

// Subtract zero from stack pointer.
SUB SP, SP, #0
STR r0, [SP]

// Subtract one from stack pointer.
SUB SP, SP, #4
STR r0, [SP]

// Subtract maximum from stack pointer.
SUB SP, SP, #508
STR r0, [SP]
STR r1, [SP, #4]
STR r2, [SP, #8]
STR r3, [SP, #12]
STR r4, [SP, #16]
STR r5, [SP, #20]
STR r6, [SP, #24]
STR r7, [SP, #28]
