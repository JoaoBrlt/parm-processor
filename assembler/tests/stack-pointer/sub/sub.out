v2.0 raw
2001 2102 2203 2304 2405 2506 2607 2708 b080 9000 b081 9000 b0ff 9000 9101 9202 9303 9404 9505 9606 9707 defd
